python/ray/data/datasource/file_based_datasource.py

```diff
@@ -251,6 +253,10 @@ def _open_input_source(
     def create_reader(self, **kwargs):
         return _FileBasedDatasourceReader(self, **kwargs)
 
+    def num_threads_per_task(self) -> int:
+        """Returns the number of threads for concurrent reading within a read task."""
+        return self._NUM_THREADS_PER_TASK
```

Comment:
This would add a new API for `FileBasedDatasource` (`@DeveloperAPI`). Shall we add later if needed? Do you think each data source needs have different values right now?

I think we can start with defining this as a constant in this file. Wanted to avoid we add API too easily.

Reply:
I'll remove this method and keep the per-class constant.

> Do you think each data source needs have different values right now?

Initially I was thinking we may want to set a smaller number for Parquet, as it's already bulk reading. But it turns out that ParquetDataSource already overrode the `get_read_tasks` function, so it's still single-threaded.

Reply:
Got it, also just realized this PR not covering ParquetDatasource.